### PR TITLE
Refine security error messages

### PR DESCRIPTION
**Pull request by craigtaverner**
Several improvements to security error messages:
- Ensure security procedures provide HasStatus exceptions
- Reduced usage of unchecked exceptions
- Added single quotes to formatted fields in messages

changelog: Improve security error messages, using correct status codes

### CODE REVIEW

**Comment by fickludd** [comment]
Good stuff!

File: community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public ReadOperations readOperations()
         if( !transaction.mode().allowsReads() )
         {
             throw new AuthorizationViolationException(
-                    String.format( "Read operations are not allowed for `%s` transactions.", transaction.mode().name() ) );
+                    String.format( "Read operations are not allowed for '%s'.", transaction.mode().name() ) );
```
Comment (henriknyman) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java:82]: Does the browser apply Markdown to the error messages? (Just curious about the previous format)

Comment (OliviaYtterbrink) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java:82]: Not that I can see.

Comment (craigtaverner) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java:82]: I also checked, and the back-ticks look like back-ticks in the browser, so no markdown support.

**Pull request merged**